ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -482,4 +487,31 @@ private void unsetProperty(final String property) {
     terminal.writer()
         .printf("Successfully unset local property '%s' (value was '%s').%n", property, oldValue);
   }
+
+  private <R> RestResponse<R> makeKsqlRequest(
+      final String ksql, final BiFunction<String, Long, RestResponse<R>> requestIssuer) {
+    return makeKsqlRequest(ksql, requestIssuer, COMMAND_QUEUE_CATCHUP_TIMEOUT_RETRIES);
+  }
+
+  private <R> RestResponse<R> makeKsqlRequest(
+      final String ksql,
+      final BiFunction<String, Long, RestResponse<R>> requestIssuer,
+      final int remainingRetries) {
+    final RestResponse<R> response =
+        requestIssuer.apply(ksql, CommandStoreUtil.LAST_SEQUENCE_NUMBER);
+    if (isSequenceNumberTimeout(response)) {
+      if (remainingRetries > 0) {
+        return makeKsqlRequest(ksql, requestIssuer, remainingRetries - 1);
+      } else {
+        return requestIssuer.apply(ksql, null);
```

Comment:
The thinking was that after we've timed out a few times, we could re-issue the request without specifying a sequence number, which defaults to no waiting (the current behavior of the CLI), so CLI users won't have to be exposed to these new timeout errors. If we'd rather expose users to the error, this can easily be removed.

Reply:
This would make the default behaviour of the CLI unsafe again though. I thought we'd agreed that the CLI, by default, should always wait for the previous command to have completed.

TBH, if after several attempts the command has still not executed, there is probably something else going on.  Just blindly firing the command off again, then likely timing out waiting for the response, (which means the user gets no feed back), probably isn't ideal.   I think a descriptive error message to the user would be more useful.